builtin/logical/transit/path_decrypt_test.go

```diff
@@ -64,74 +65,155 @@ func TestTransit_BatchDecryption(t *testing.T) {
 }
 
 func TestTransit_BatchDecryption_DerivedKey(t *testing.T) {
+	var req *logical.Request
 	var resp *logical.Response
 	var err error
 
 	b, s := createBackendWithStorage(t)
 
-	policyData := map[string]interface{}{
-		"derived": true,
-	}
-
-	policyReq := &logical.Request{
+	// Create a derived key.
+	req = &logical.Request{
 		Operation: logical.UpdateOperation,
 		Path:      "keys/existing_key",
 		Storage:   s,
-		Data:      policyData,
+		Data: map[string]interface{}{
+			"derived": true,
+		},
 	}
-
-	resp, err = b.HandleRequest(context.Background(), policyReq)
+	resp, err = b.HandleRequest(context.Background(), req)
 	if err != nil || (resp != nil && resp.IsError()) {
 		t.Fatalf("err:%v resp:%#v", err, resp)
 	}
 
-	batchInput := []interface{}{
-		map[string]interface{}{"plaintext": "dGhlIHF1aWNrIGJyb3duIGZveA==", "context": "dGVzdGNvbnRleHQ="},
-		map[string]interface{}{"plaintext": "dGhlIHF1aWNrIGJyb3duIGZveA==", "context": "dGVzdGNvbnRleHQ="},
+	// Encrypt some values for use in test cases.
+	plaintextItems := []BatchRequestItem{
```

Comment:
Not that it matters a whole lot, but instead of using `BatchRequestItem` couldn't you simply use an anonymous struct here, as the only fields that really matter for the test are the plaintext an the context?

Reply:
Ah, so the reason I did this is mainly for convenience. In order for the encrypt request to work, the `batch_input` field has to be specifically a `[]interface{}`, which precludes map indexing without type assertions to a `map[string]interface{}`. I just made this as kind of a source of truth lookup table for easy test definitions. It lets me define test input and output without duplicating the strings inline in each test.

Reply:
I agree that using a struct is a good idea. I was suggesting using an anonymous struct rather than `BatchRequestItem` since the latter has many more fields that are not really relevant to the tests.

Something like:
```
plaintextItems := []struct{
    plaintext, context string
}{
...
```

Reply:
Gotcha, I see what you mean now. I can certainly use an inline struct here to make the the data more directly relevant to tests. Thanks for explaining that!